restore/data.go

```diff
@@ -215,34 +215,49 @@ func restoreDataFromTimestamp(fpInfo filepath.FilePathInfo, dataEntries []toc.Co
 		gplog.Verbose("Initializing pipes and gpbackup_helper on segments for %srestore", msg)
 		utils.VerifyHelperVersionOnSegments(version, globalCluster)
 
-		// During a larger-to-smaller restore, we need to do multiple passes of
-		// data loading so we assign the batches here.
-		oidList := make([]string, 0)
-		for _, entry := range dataEntries {
-			if entry.IsReplicated {
-				oidList = append(oidList, fmt.Sprintf("%d,0", entry.Oid))
-				continue
-			}
+		var maxHelpers int
+		if backupConfig.SingleDataFile {
+			maxHelpers = 1
+		} else if connectionPool.NumConns < totalTables {
+			maxHelpers = connectionPool.NumConns
+		} else {
+			maxHelpers = totalTables
+		}
+
+		for j := 0; j < maxHelpers; j++ {
```

Comment:
Nice to have: make `j` more informative, for instance, `helperIdx`.

Reply:
renamed